lib/file.js

```diff
@@ -2,128 +2,123 @@ const fs = require('fs')
-        const icns = info.CFBundleIconFile.substr(-5) === '.icns' ? info.CFBundleIconFile : `${info.CFBundleIconFile}.icns`
-        const iconPath = path.join(this.path, 'contents', 'Resources', icns)
-        const getSize = (name) => name.match(/\d+/)[0]
-        iconutil.toIconset(iconPath, (err, icons) => {
-          if (err) return reject(err)
-          const biggestIcon = Object.keys(icons).reduce((biggest, latest) => {
-            const biggestSize = getSize(biggest)
-            const latestSize = getSize(latest)
-            return biggestSize > latestSize ? biggest : latest
-          }, 'icon_0x0.png')
-          fs.writeFile('Zazu.app.png', icons[biggestIcon], 'binary', (err) => {
-            if (err) throw err
-            console.log('File saved.')
-          })
+        return info.CrAppModeShortcutName || info.CFBundleDisplayName || info.CFBundleName || fileName
+    }
+
+    isDirectory() {
+        const isDirectory = this.stats.isDirectory()
+        const isSymbolicLink = this.stats.isSymbolicLink()
+        return isDirectory && !isSymbolicLink && !this.isAppMac()
+    }
+
+    isBroken() {
+        return !this.stats
+    }
+
+    relativePath() {
+        return this.path.replace(HOME_REGEX, '~')
+    }
+
+    info() {
+        try {
+            const infoPath = path.join(this.path, 'contents', 'Info.plist')
+            return plist.parse(fs.readFileSync(infoPath).toString())
+        } catch (e) {
+            return {}
+        }
+    }
+
+    toJson() {
+        return {
+            icon: this.isDirectory() ? 'fa-folder' : getIcon(this.relativePath()),
+            title: this.title(),
+            subtitle: this.relativePath(),
+            value: this.relativePath(),
+            id: this.relativePath(),
+        }
+    }
+
+    getStats () {
+        return new Promise((resolve, reject) => {
+            fs.stat(this.path, (err, stats) => {
+                if (!err) this.stats = stats
+                resolve()
+            })
         })
-      }
-      resolve(this.isDirectory() ? 'fa-folder' : 'fa-file')
-    })
-  }
-
-  toJson () {
-    return new Promise((resolve) => {
-      return this.icon()
-    }).then((icon) => {
-      return {
-        icon,
-        title: this.title(),
-        subtitle: this.relativePath(),
-        value: this.relativePath(),
-        id: this.relativePath(),
-      }
-    })
-  }
-
-  getStats () {
-    return new Promise((resolve, reject) => {
-      fs.stat(this.path, (err, stats) => {
-        if (!err) this.stats = stats
-        resolve()
-      })
-    })
-  }
+    }
 }
 
 module.exports = File
+
+function getIcon(path) {
+    let iconImg = 'fa-file';
+
+    // TODO Refactoring & Test
+    if (path[0] === '~') {
+        path = path.replace('~', process.env.HOME);
+    }
+    if (path.split('.').indexOf('app') >= 0 && fs.existsSync(path)) {
+        const fs = require('fs');
+        fs.readdirSync(path + '/Contents/Resources/').forEach(file => {
```

Comment:
This is the first prototype. The next step is reading Info.plist and get the icon file. Do you want the next step?